server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java

```diff
@@ -359,6 +359,22 @@ private static void updateOneTargetMods(
     targetFile.getModFile().close();
   }
 
+  public static void removeCompactionModification(
+      List<TsFileResource> selectedSeqTsFileResourceList,
+      List<TsFileResource> selectedUnSeqTsFileResourceList,
+      String fullStorageGroupName) {
+    try {
+      for (TsFileResource seqFile : selectedSeqTsFileResourceList) {
+        ModificationFile.getCompactionMods(seqFile).remove();
```

Comment:
How about making getCompactionMods() as a method of TsFileResource

Reply:
resolved